google/resource_iam_binding.go

```diff
@@ -82,6 +82,11 @@ func resourceIamBindingRead(newUpdaterFunc newResourceIamUpdaterFunc) schema.Rea
 		eBinding := getResourceIamBinding(d)
 		p, err := updater.GetResourceIamPolicy()
 		if err != nil {
+			if isGoogleApiErrorWithCode(err, 404) {
+				log.Printf("[DEBUG]: Binding for role %q not found for non-existant resource %s, removing from state file.\n", updater.DescribeResource(), eBinding.Role)
```

Comment:
If you want to remove it from the state file, you'll also want to add `d.SetId("")`.
Also I realize there are other log statements with it in here but I don't think the extra `\n` is really necessary